[CI] QueryProfilerIT testProfileMatchesRegular is failing

Test failure: https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+intake/2432/console

fails with:

```
QueryProfilerIT.testProfileMatchesRegular <<< FAILURES!
> Throwable #1: java.lang.AssertionError: Profile totalHits of [127] is not close to Vanilla totalHits [129]
> Expected: <127L>
>      but: was <129L>
> 	at __randomizedtesting.SeedInfo.seed([32784220547E92F2:61AD95220536708E]:0)
> 	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
> 	at org.elasticsearch.search.profile.query.QueryProfilerIT.testProfileMatchesRegular(QueryProfilerIT.java:162)
> 	at java.lang.Thread.run(Thread.java:748)
```

Reproduction line:

```
./gradlew :server:integTest \
  -Dtests.seed=32784220547E92F2 \
  -Dtests.class=org.elasticsearch.search.profile.query.QueryProfilerIT \
  -Dtests.method="testProfileMatchesRegular" \
  -Dtests.security.manager=true \
  -Dtests.locale=id \
  -Dtests.timezone=America/Curacao
```

(does not reproduce locally)

Another one in master

https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+multijob-unix-compatibility/os=centos/33/console

[build-master-centos-33.txt.zip](https://github.com/elastic/elasticsearch/files/2540520/build-master-centos-33.txt.zip)

```
  2> REPRODUCE WITH: ./gradlew :server:integTest -Dtests.seed=91857F7F3931F7A9 -Dtests.class=org.elasticsearch.search.profile.query.QueryProfilerIT -Dtests.method="testProfileMatchesRegular" -Dtests.security.manager=true -Dtests.locale=en-AU -Dtests.timezone=Etc/GMT-0 -Dcompiler.java=11 -Druntime.java=8
```

```
FAILURE 0.34s J0 | QueryProfilerIT.testProfileMatchesRegular <<< FAILURES!
   > Throwable #1: java.lang.AssertionError: Vanilla hits were larger than profile hits.  Non-overlapping elements were: [{
   >   "_index" : "test",
   >   "_type" : "type1",
   >   "_id" : "106",
   >   "_score" : null,
   >   "_source" : {
   >     "field1" : "one hundred six ",
   >     "field2" : 106
   >   },
   >   "sort" : [
   >     "106"
   >   ]
   > }, {
   >   "_index" : "test",
   >   "_type" : "type1",
   >   "_id" : "104",
   >   "_score" : null,
   >   "_source" : {
   >     "field1" : "one hundred four ",
   >     "field2" : 104
   >   },
   >   "sort" : [
   >     "104"
   >   ]
   > }, {
   >   "_index" : "test",
   >   "_type" : "type1",
   >   "_id" : "16",
   >   "_score" : null,
   >   "_source" : {
   >     "field1" : "sixteen ",
   >     "field2" : 16
   >   },
   >   "sort" : [
   >     "16"
   >   ]
   > }, {
   >   "_index" : "test",
   >   "_type" : "type1",
   >   "_id" : "102",
   >   "_score" : null,
   >   "_source" : {
   >     "field1" : "one hundred two ",
   >     "field2" : 102
   >   },
   >   "sort" : [
   >     "102"
   >   ]
   > }, {
   >   "_index" : "test",
   >   "_type" : "type1",
   >   "_id" : "13",
   >   "_score" : null,
   >   "_source" : {
   >     "field1" : "thirteen ",
   >     "field2" : 13
   >   },
   >   "sort" : [
   >     "13"
   >   ]
   > }, {
   >   "_index" : "test",
   >   "_type" : "type1",
   >   "_id" : "17",
   >   "_score" : null,
   >   "_source" : {
   >     "field1" : "seventeen ",
   >     "field2" : 17
   >   },
   >   "sort" : [
   >     "17"
   >   ]
   > }, {
   >   "_index" : "test",
   >   "_type" : "type1",
   >   "_id" : "11",
   >   "_score" : null,
   >   "_source" : {
   >     "field1" : "eleven ",
   >     "field2" : 11
   >   },
   >   "sort" : [
   >     "11"
   >   ]
   > }, {
   >   "_index" : "test",
   >   "_type" : "type1",
   >   "_id" : "103",
   >   "_score" : null,
   >   "_source" : {
   >     "field1" : "one hundred three ",
   >     "field2" : 103
   >   },
   >   "sort" : [
   >     "103"
   >   ]
   > }, {
   >   "_index" : "test",
   >   "_type" : "type1",
   >   "_id" : "14",
   >   "_score" : null,
   >   "_source" : {
   >     "field1" : "fourteen ",
   >     "field2" : 14
   >   },
   >   "sort" : [
   >     "14"
   >   ]
   > }, {
   >   "_index" : "test",
   >   "_type" : "type1",
   >   "_id" : "10",
   >   "_score" : null,
   >   "_source" : {
   >     "field1" : "ten ",
   >     "field2" : 10
   >   },
   >   "sort" : [
   >     "10"
   >   ]
   > }]
   >    at __randomizedtesting.SeedInfo.seed([91857F7F3931F7A9:C250A87D687915D5]:0)
   >    at org.elasticsearch.search.profile.query.QueryProfilerIT.testProfileMatchesRegular(QueryProfilerIT.java:173)
   >    at java.lang.Thread.run(Thread.java:748)
```

--- a/server/src/test/java/org/elasticsearch/search/profile/query/QueryProfilerIT.java
+++ b/server/src/test/java/org/elasticsearch/search/profile/query/QueryProfilerIT.java
@@ -26,6 +26,7 @@ import org.elasticsearch.action.search.SearchRequestBuilder;
 import org.elasticsearch.action.search.SearchResponse;
 import org.elasticsearch.action.search.SearchType;
 import org.elasticsearch.action.search.ShardSearchFailure;
+import org.elasticsearch.common.settings.Settings;
 import org.elasticsearch.index.query.QueryBuilder;
 import org.elasticsearch.index.query.QueryBuilders;
 import org.elasticsearch.search.SearchHit;
@@ -110,7 +111,9 @@ public class QueryProfilerIT extends ESIntegTestCase {
      * to make sure the profiling doesn't interfere with the hits being returned
      */
     public void testProfileMatchesRegular() throws Exception {
-        createIndex("test");
+        createIndex("test", Settings.builder() 
+                .put("index.number_of_shards", 1)
+                .put("index.number_of_replicas", 0).build());
         ensureGreen();
 
         int numDocs = randomIntBetween(100, 150);

